src/main/kotlin/domain/CarRacingGame.kt

```diff
@@ -0,0 +1,45 @@
+package domain
+
+import data.Car
+import data.CarPath
+import data.PathState
+import data.generator.RacingNumberGenerator
+
+class CarRacingGame {
+
+    fun startDriving(cars: List<Car>, numberOfTry: Int): List<CarPath> {
+        val carsPath = mutableListOf<CarPath>()
+
+        cars.forEach { car ->
+            carsPath.add(recordPath(car, numberOfTry))
+        }
+
+        return carsPath.toList()
+    }
+
+    private fun recordPath(car: Car, numberOfTry: Int): CarPath {
+        val path = mutableListOf<PathState>()
+
+        repeat(numberOfTry) {
+            val randomNumber = RacingNumberGenerator().generate()
+            path.add(car.move(randomNumber))
+        }
+
+        return CarPath(car.name, path)
+    }
+
+    fun decideWinner(cars: List<Car>): List<String> {
+        val maxScore = cars.getMaxScore()
+        val winners = cars.filter { car ->
+            car.position == maxScore
+        }.map { car ->
+            car.name
+        }
```

Comment:
메서드 체이닝은 이런 형태로 만들어 보는 것은 어떨까요?
```suggestion
        val winners = cars.filter { car -> car.position == maxScore }
            .map { car -> car.name }
```